src/internal/api/packages/find.go

```diff
@@ -5,52 +5,177 @@
 package packages
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/fs"
 	"net/http"
 	"os"
+	"path/filepath"
 	"regexp"
 
-	"github.com/defenseunicorns/zarf/src/internal/api/common"
-	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/pkg/packager"
 	"github.com/defenseunicorns/zarf/src/pkg/utils"
 )
 
 // Find zarf-packages on the local system (https://regex101.com/r/TUUftK/1)
 var packagePattern = regexp.MustCompile(`zarf-package[^\s\\\/]*\.tar(\.zst)?$`)
-// Find zarf-init packages on the local system (https://regex101.com/r/6aTl3O/2)
-var initPattern = regexp.MustCompile(`zarf-init[^\s\\\/]*\.tar(\.zst)?$`)
 
-// Find returns all packages anywhere down the directory tree of the working directory.
-func Find(w http.ResponseWriter, _ *http.Request) {
-	message.Debug("packages.Find()")
-	findPackage(packagePattern, w, os.Getwd)
+// Find zarf-init packages on the local system
+var currentInitPattern = regexp.MustCompile(packager.GetInitPackageName(""))
+
+// FindInHomeStream returns all packages in the user's home directory.
+// If the init query parameter is true, only init packages will be returned.
+func FindInHomeStream(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/event-stream; charset=utf-8")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Connection", "keep-alive")
+
+	init := r.URL.Query().Get("init")
+	regexp := packagePattern
+	if init == "true" {
+		regexp = currentInitPattern
+	}
+
+	done := make(chan bool)
+	go func() {
+		// User home directory
+		homePath, err := os.UserHomeDir()
+		if err != nil {
+			streamError(err, w)
+		} else {
+			// Recursively search for and stream packages in the home directory
+			recursivePackageStream(homePath, regexp, w)
+		}
+		close(done)
+	}()
+
+	<-done
 }
 
-// FindInHome returns all packages in the user's home directory.
-func FindInHome(w http.ResponseWriter, _ *http.Request) {
-	message.Debug("packages.FindInHome()")
-	findPackage(packagePattern, w, os.UserHomeDir)
+// FindInitStream finds and streams all init packages in the current working directory, the cache directory, and execution directory
+func FindInitStream(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "text/event-stream; charset=utf-8")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Connection", "keep-alive")
+
+	done := make(chan bool)
+	go func() {
+		// stream init packages in the execution directory
+		if execDir, err := os.Getwd(); err == nil {
+			streamDirPackages(execDir, currentInitPattern, w)
+		} else {
+			streamError(err, w)
+		}
```

Comment:
This doesn't get the binary location. See `GetFinalExecutablePath()` in `src/pkg/utils/io.go`

Reply:
Yep I missed that when I was moving things around.